### PR TITLE
Feature/1502 search filtering

### PR DESCRIPTION
**Pull request by angtheod**
Added backend and frontend functionality for selecting filters and show the selected ones properly both in page and modal mode. Added filtering of input values that will be used for the actual filtering of the results.

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -1678,6 +1678,10 @@ body { background: #b0d4c8; font-family: "Roboto", sans-serif; }
 
 body.brown-bg { background: #FAF7EC; }
 
+.bold { font-weight: bold; }
+.activefilter-tag { background-color: transparent; text-align: left; cursor: pointer; }
+.clearall { cursor: pointer; text-transform: uppercase; }
+
```
Comment (angtheod) [inline style.css]: @ashish173 
Ashish I am tagging you at the lines that you need to see for adding changes in presentation layer.

File: templates/search.twig [hunk 2]
```diff
@@ -163,11 +163,11 @@
...
 									<div class="activefilter-list">
 										{% for filter in filters %}
-											<a href="{{ filter.link }}" class="activefilter-tag">
+											<button class="activefilter-tag" data-id="{{ filter.id }}">
 												{{ filter.name }} <i class="icon-cross"></i>
-											</a>
+											</button>
 										{% endfor %}
-										<a href="#" class="clearall">{{ __( 'Clear All', domain ) }} <i class="icon-cross"></i></a>
+										<button class="clearall">{{ __( 'Clear All', domain ) }} <i class="icon-cross"></i></button>
```
Comment (angtheod) [inline templates/search.twig]: @ashish173 
Changed these anchors to buttons.

File: assets/js/main.js [hunk 3]
```diff
@@ -23,6 +23,38 @@ $(document).ready(function() {
...
+		$(".custom-control-input[value=" + $(this).data("id") + "]").prop("checked", false);
+		$("#search_form").submit();
+	});
+
+	// Clear all selected filters.
+	$(".clearall").off("click").on("click", function () {
+		$("input[name^='f[']").prop("checked", false);
+		$("#search_form").submit();
+	});
+
```
Comment (angtheod) [inline assets/js/main.js]: @ashish173 
Not sure, if you need something from here but these are the js additions

File: templates/search.twig [hunk 4]
```diff
@@ -136,9 +136,9 @@
 																{% for tag in tags %}
 																	<li>
 																		<label class="custom-control custom-checkbox">
-																			<input type="checkbox" class="custom-control-input" {{ disabled }} />
+																			<input type="checkbox" name="f[tag][{{ tag.name }}]" value="{{ tag.term_id }}" class="custom-control-input modal-checkbox"{{ tag.checked }} {{ disabled }} />
```
Comment (angtheod) [inline templates/search.twig]: @ashish173 
Needed to add a class only for the checkbox's found in the modal.

File: classes/class-p4-search.php [hunk 5]
```diff
@@ -189,26 +188,88 @@ protected function add_results_context( &$context ) {
...
+		}
+
+		/**
+		 * Validates the input.
+		 *
+		 * @param  int $id The id to be validated.
+		 *
+		 * @return bool True if validation is ok, false if validation fails.
+		 */
+		public function validate( $id ) : bool {
```
Comment (sagarsdeshmukh) [inline classes/class-p4-search.php]: Do you think we need a separate function to validate ID ? We can use `filter_var($int, FILTER_VALIDATE_INT))` 
}

**Comment by ashish173** [comment]
@angtheod okay, thanks! I think we need to get on a call maybe later today for this one.

**Pull request merged**